src/dags/dag_tree.py

```diff
@@ -0,0 +1,372 @@
+            parameter_path = _link_parameter_to_function_or_input(
+                flat_functions, namespace, parameter_name, namespace_of_inputs
+            )
+
+            if parameter_path not in flat_functions:
+                flat_input_structure[parameter_path] = None
+
+    nested_input_structure = _unflatten_str_dict(flat_input_structure)
+
+    # If no targets are specified, all inputs are needed
+    if targets is None:
+        return nested_input_structure
+
+    # Compute transitive hull of inputs needed for given targets
+    flat_renamed_functions = _flatten_functions_and_rename_parameters(
+        functions, nested_input_structure
+    )
+    flat_targets = _flatten_targets(targets)
+    dag = create_dag(flat_renamed_functions, flat_targets)
+    parameters = _create_arguments_of_concatenated_function(flat_renamed_functions, dag)
+
+    return _unflatten_str_dict({parameter: None for parameter in parameters})
+
+
+def _flatten_str_dict(str_dict: NestedStrDict) -> FlatStrDict:
+    return flatten(str_dict, reducer=_qualified_name_reducer)
+
+
+def _unflatten_str_dict(str_dict: FlatStrDict) -> NestedStrDict:
+    return unflatten(str_dict, splitter=_qualified_name_splitter)
+
+
+def _flatten_targets(targets: Optional[NestedTargetDict]) -> Optional[FlatTargetList]:
+    if targets is None:
+        return None
+
+    return list(_flatten_str_dict(targets).keys())
+
+
+def _link_parameter_to_function_or_input(
+    flat_functions: FlatFunctionDict,
+    namespace: str,
+    parameter_name: str,
+    namespace_of_inputs: GlobalOrLocal = "local",
+) -> str:
+    """
+    Returns the path to the function/input that the parameter points to.
+
+    If the parameter name has double underscores (e.g. "namespace1__f"), we know it
+    represents a qualified name and the path simply consists of the segments of the
+    qualified name (e.g. "namespace1, "f").
+
+    Otherwise, we cannot be sure whether the parameter points to a function/input of
+    the current namespace or a function/input of the top level. In this case, we
+        (1) look for a function with that name in the current namespace,
+        (2) look for a function with that name in the top level, and
+        (3) assume the parameter points to an input.
+    In the third case, `namespace_of_inputs` determines whether the parameter points
+    to an input of the current namespace ("local") or an input of the top level
+    ("global").
+
+    Args:
+        flat_functions:
+            The flat dictionary of functions.
+        namespace:
+            The namespace that contains the function that contains the parameter.
+        parameter_name:
+            The name of the parameter.
+        namespace_of_inputs:
+            The level of inputs to assume if the parameter name does not represent a
+            function.
+    Returns:
+        The path to the function/input that the parameter points to.
+    """
+
+    # Parameter name is definitely a qualified name
```

Comment:
Does this work with nesting > 2 levels?  At a quick glance I did not see a test with 3 levels, but I need to stop now.

But maybe we decided that we only want to have 2 levels and I forgot (and then there should be a big red flag in the documentation that I missed).